pkg/api/converterfromapi.go

```diff
@@ -182,10 +190,18 @@ func convertPropertiesToV20160330(api *Properties, p *v20160330.Properties) {
 		convertMasterProfileToV20160330(api.MasterProfile, p.MasterProfile)
 	}
 	p.AgentPoolProfiles = []*v20160330.AgentPoolProfile{}
-	for _, apiProfile := range api.AgentPoolProfiles {
-		v20160330Profile := &v20160330.AgentPoolProfile{}
-		convertAgentPoolProfileToV20160330(apiProfile, v20160330Profile)
-		p.AgentPoolProfiles = append(p.AgentPoolProfiles, v20160330Profile)
+	// We expect exactly 2 agent pools: index [0] is the private pool, index [1] is the public pool
+	for i, apiProfile := range api.AgentPoolProfiles {
+		// a V20160330 should only have one (private) agent pool
+		if i < 1 {
+			v20160330Profile := &v20160330.AgentPoolProfile{}
```

Comment:
I know this will work based on validation but is there a way to detect which is public/private agent pools outside of just going off of the index?  I'm thinking about a future where a cluster is created with version v20170701 then someone calls get with v20170131 afterwards. Will this behave well